code/game/objects/items/devices/pager.dm

```diff
@@ -0,0 +1,148 @@
+				[prefs["Fire"] == "loud" ? "<b>Loud</b>" : {"<a href="?src=\ref[src];fire_prefs=loud">Loud</a>"}]
+			<li><b>Power Alerts: </b>
+				[prefs["Power"] == "silent" ? "<b>Silent</b>" : {"<a href="?src=\ref[src];power_prefs=silent">Silent</a>"}]
+				[prefs["Power"] == "quiet" ? "<b>Quiet</b>" 	 : {"<a href="?src=\ref[src];power_prefs=quiet">Quiet</a>"}]
+				[prefs["Power"] == "loud" ? "<b>Loud</b>" : {"<a href="?src=\ref[src];power_prefs=loud">Loud</a>"}]
+			</li>
+		</ul>"}
+	dat += "Silent will provide no audible indication of an alert.<BR>"
+	dat += "Quiet will announce an alert with no sound effects.<BR>"
+	dat += "Loud will announce an alert and play a sound effect.<BR><BR>"
+	dat += "Pager status: [muted ? "<A href='byond://?src=\ref[src];toggle_mute'>Muted</A>" : "<A href='byond://?src=\ref[src];toggle_mute'>Active</A>"]<BR>"
+
+	dat = jointext(dat,"")
+	var/datum/browser/popup = new(user, "\ref[src]", name, 400, 500)
+	popup.set_content(dat)
+	popup.open()
+	return
+
+/obj/item/device/pager/Topic(href, href_list)
+	if(!in_range(src,usr) && !isAdminGhost(usr) && !issilicon(usr))
+		usr << browse(null, "window=pager")
+		return 1
+	usr.set_machine(src)
+	if("toggle_mute" in href_list)
+		toggle_mute()
+	else if ("atmos_prefs" in href_list)
+		prefs["Atmosphere"] = href_list["atmos_prefs"]
+	else if ("fire_prefs" in href_list)
+		prefs["Fire"] = href_list["fire_prefs"]
+	else if ("power_prefs" in href_list)
+		prefs["Power"] = href_list["power_prefs"]
+	if (!( master ))
+		if (istype(loc, /mob))
+			interact(loc)
+		else
+			updateDialog()
+	else
+		if (istype(master.loc, /mob))
+			interact(master.loc)
+		else
+			updateDialog()
+	add_fingerprint(usr)
+
+/obj/item/device/pager/AltClick(mob/user)
+	if(!(user) || !isliving(user))
+		return FALSE
+	if(user.incapacitated() || !Adjacent(user))
+		return FALSE
+	toggle_mute(user)
+
+/obj/item/device/pager/proc/toggle_mute(mob/user)
+	if(muted)
+		muted = FALSE
+		to_chat(usr, "<span class = 'caution'>You unmute \the [src].</span>")
+		update_icon()
+	else
+		muted = TRUE
+		to_chat(usr, "<span class = 'caution'>You mute \the [src].</span>")
+		update_icon()
+
+/obj/item/device/pager/proc/triggerAlarm(var/class, area/A)
+	if(muted)
+		return
+	var/alarmlevel = prefs[class]
+	var/alarmtext = "[class] alarm detected in \the [A]!"
+	if(alarmtext == last_alert)
+		return
+	last_alert = alarmtext
+	var/sleeptime = rand(0,3)
+	sleep(sleeptime SECONDS) //provides some variety in when the pagers are triggered rather than all firing simultaneously
+	switch(alarmlevel)
+		if("silent")
+			icon_state = "pager_active"
+			sleep(3 SECONDS)
+			update_icon()
+		if("quiet")
+			icon_state = "pager_active"
+			say(alarmtext)
+			sleep(3 SECONDS)
+			update_icon()
+		if("loud")
+			icon_state = "pager_active"
+			say(alarmtext)
+			if(world.time - last_alert_time >= alert_delay)
+				playsound(src, 'sound/effects/3beep.ogg', 100, 0, 1)
+				last_alert_time = world.time
+			sleep(3 SECONDS)
```

Comment:
seems to me that icon_state = active could come before the switch
whereas sleep(3 seconds) and update icon could come after the switch
thus eliminating 6 lines